src/app/services/xml-parsers/basic-parsers.ts

```diff
@@ -329,3 +424,141 @@ export class TermParser extends GenericElemParser implements Parser<XMLElement>
         };
     }
 }
+
+@xmlParser('milestone', MilestoneParser)
+export class MilestoneParser extends GenericElemParser implements Parser<XMLElement> {
+    parse(xml: XMLElement): Milestone {
+
+        const endElement = (xml.getAttribute('spanTo')) ? getExternalElements(xml, ['spanTo'], 'xml:id', 'anchor') : [];
+        const includedElements = (endElement.length !== 0) ? getElementsBetweenTreeNode(xml, endElement[0]) : [];
+        const parsedElements = (includedElements.length !== 0) ?
+            includedElements.map((x: XMLElement) => (x.nodeType !== 3 && x.nodeType !== 8) ? super.parse(x) : x) : [];
+
+        return {
+            type: Milestone,
+            id: xml.getAttribute('xml:id'),
+            attributes: this.attributeParser.parse(xml),
+            unit: xml.getAttribute('unit'),
+            spanText: '',
+            spanElements: parsedElements,
+            content: parseChildren(xml, this.genericParse),
+        };
+    }
+}
+
+@xmlParser('anchor', AnchorParser)
+export class AnchorParser extends GenericElemParser implements Parser<XMLElement> {
+
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    //todo: check if a span is referring to this element's @xml:id?
+
+    parse(xml: XMLElement): Anchor {
+
+        return {
+            type: Anchor,
+            id: xml.getAttribute('xml:id'),
+            attributes: this.attributeParser.parse(xml),
+            content: parseChildren(xml, this.genericParse),
+        };
+    }
+}
+
+
+@xmlParser('spanGrp', SpanParser)
+@xmlParser('span', SpanParser)
+export class SpanParser extends GenericElemParser implements Parser<XMLElement> {
+
+    attributeParser = createParser(AttributeParser, this.genericParse);
+
+    parse(xml: XMLElement): Span|SpanGrp {
+
+        if (xml.tagName === 'spanGrp') {
+
+            return <SpanGrp> {
+                type: SpanGrp,
+                id: xml.getAttribute('xml:id'),
+                attributes: this.attributeParser.parse(xml),
+                spans: Array.from(xml.querySelectorAll<XMLElement>('span')).map((x) => <Span>this.parse(x)),
+                content: parseChildren(xml, this.genericParse),
+            }
+
+        } else if (xml.tagName === 'span') {
```

Comment:
I find it less readable because if it's the first if is verified then cannot be verified the second if (and so it should be a else-if).
Anyway I'm changing it as you requested